candle-core/Cargo.toml

```diff
@@ -48,4 +48,4 @@ metal = ["dep:metal", "dep:candle-metal-kernels"]
 [[bench]]
 name = "matmul"
 harness = false
-
+required-features = ["metal"]
```

Comment:
I don't think we want metal by default here.

Reply:
The [benchmark calls new_metal](https://github.com/huggingface/candle/blob/fa3ea98ba92835960fdd825a5b4dda30ef2baaa4/candle-core/benches/matmul.rs#L15) which is [only available when the metal feature is enabled](https://github.com/huggingface/candle/blob/fa3ea98ba92835960fdd825a5b4dda30ef2baaa4/candle-core/src/lib.rs#L92-L94).

Reply:
Yes, and we'll have to fix that.
I have a branch locally where I conditionally load the correct benchmarks from different modules via `cfg`.
I can make a PR tomorrow.

Reply:
Okay I figured that was the original intention. FWIW, rust-analyzer treats it as an error currently.

> error[E0599]: no method named `wait_until_completed` found for reference `&MetalDevice` in the current scope
  --> candle-core/benches/matmul.rs:31:24
   |
31 |                 device.wait_until_completed().unwrap();
   |                        ^^^^^^^^^^^^^^^^^^^^ method not found in `&MetalDevice`

Reply:
Something like this?
https://github.com/huggingface/candle/pull/1538

Reply:
I kind of agree with should keep this out. Also the bench will be useful for other targets other than metal.

Reply:
okay it has been removed from the PR!